Add a brief explanation of the Task Asynchronous Protocol (#39181)
dotnet/docs: docs/csharp/tour-of-csharp/features.md

- The body of the method `await`s the return of the `GetByteArrayAsync` method.
- The type specified in the `return` statement matches the type argument in the `Task<T>` declaration for the method. (A method that returns a `Task` would use `return` statements without any argument).

> The [`async`](../language-reference/keywords/async.md) and [`await`](../language-reference/operators/await.md) provide a language-level abstraction over lower-level primitives that support asynchronous operations. The compiler and library may rely on network interrupts, OS events, hardware interrupts, or other primitives to trigger the completion of an asynchronous operation. The compiler generates code to update the state of any <xref:System.Threading.Tasks.Task?displayProperty=nameWithType> object and transfer control to code that should execute when the awaited task completes.

Reviewer: ```suggestion
The [`async`](../language-reference/keywords/async.md) keyword and [`await`](../language-reference/operators/await.md) operator provide a language-level abstraction over lower-level primitives that support asynchronous operations. The compiler and library might rely on network interrupts, OS events, hardware interrupts, or other primitives to trigger the completion of an asynchronous operation. The compiler generates code to update the state of any <xref:System.Threading.Tasks.Task> object and transfer control to code that should execute when the awaited task completes.
```
